nexus/db-queries/src/db/datastore/sled.rs

```diff
@@ -194,6 +201,14 @@ impl DataStore {
             .await
             .map_err(|e| match e {
                 TxnError::CustomError(SledReservationError::NotFound) => {
+                    // XXX: Should this be a different error? 503 Service
+                    // Unavailable is often for transient errors, and this
+                    // message isn't exposed to users (which seems wrong?)
+                    //
+                    // Maybe 409 Conflict as documented at
+                    // https://stackoverflow.com/a/11093566, since this request
+                    // if completed would put the system into an inconsistent
+                    // state.
```

Comment:
I think my main remaining concern is that the error message doesn't make its way to clients -- for something like this it seems like it would be useful.

Reply:
Going to turn this into a separate discussion.

Reply:
https://github.com/oxidecomputer/omicron/pull/4573